pandas/_typing.py

```diff
@@ -85,6 +85,20 @@
 PandasScalar = Union["Period", "Timestamp", "Timedelta", "Interval"]
 Scalar = Union[PythonScalar, PandasScalar]
 
+# numpy compatible types
+NumpyValueArrayLike = Union[PythonScalar, Sequence[PythonScalar]]
```

Comment:
The typing definition of `Sequence` does not include numpy arrays, EAs, Series, Index etc. #28770

from numpy types

```
    @overload
    def searchsorted(  # type: ignore[misc]
        self,  # >= 1D array
        v: _ScalarLike_co,  # 0D array-like
        side: _SortSide = ...,
        sorter: Optional[_ArrayLikeInt_co] = ...,
    ) -> intp: ...
    @overload
    def searchsorted(
        self,  # >= 1D array
        v: ArrayLike,
        side: _SortSide = ...,
        sorter: Optional[_ArrayLikeInt_co] = ...,
    ) -> ndarray[Any, dtype[intp]]: ...

```

```suggestion
NumpyValueArrayLike = npt.ArrayLike | npt._ScalarLike_co
```

Reply:
Next commit includes that change.